grapher/controls/Controls.tsx

```diff
@@ -213,48 +213,48 @@ export interface FooterControlsManager extends ShareMenuManager {
 export class FooterControls extends React.Component<{
     manager: FooterControlsManager
 }> {
-    @computed private get manager() {
+    @computed private get manager(): FooterControlsManager {
         return this.props.manager
     }
 
-    @action.bound onShareMenu() {
+    @action.bound onShareMenu(): void {
         this.manager.isShareMenuActive = !this.manager.isShareMenuActive
     }
 
-    @computed private get availableTabs() {
+    @computed private get availableTabs(): GrapherTabOption[] {
         return this.manager.availableTabs || []
     }
 
-    private _getTabsElement() {
+    private _getTabsElement(): JSX.Element {
         const { manager } = this
         return (
             <nav className="tabs">
                 <ul>
-                    {this.availableTabs.map((tabName) => {
-                        return (
-                            tabName !== GrapherTabOption.download && (
-                                <li
-                                    key={tabName}
-                                    className={
-                                        "tab clickable" +
-                                        (tabName === manager.currentTab
-                                            ? " active"
-                                            : "")
+                    {this.availableTabs.map((tabName): JSX.Element | null => {
+                        return tabName !== GrapherTabOption.download ? (
+                            <li
+                                key={tabName}
+                                className={
+                                    "tab clickable" +
+                                    (tabName === manager.currentTab
+                                        ? " active"
+                                        : "")
+                                }
+                            >
+                                <a
+                                    onClick={():
+                                        | GrapherTabOption.chart
+                                        | GrapherTabOption.map
+                                        | GrapherTabOption.sources
+                                        | GrapherTabOption.table =>
```

Comment:
I was very confused about this for a few seconds, and don't really like this extreme case.